### PR TITLE
Fix/adaptation for new search

### PR DESCRIPTION
**Pull request by kilativ-dotcom**

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - README
 - Changelog
 - Add always generating for conclusion in implication
+- Add arguments to sub formulas in conjunction and disjunction
```
Comment (MksmOrlov) [inline docs/changelog.md:22]:
```suggestion
- Add arguments to subformulas in conjunction and disjunction
```

File: docs/changelog.md [hunk 2]
```diff
@@ -28,3 +29,4 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Rename variables, log message concepts and methods according to the Standard concepts
 - Separate tests for simple formulas, complex formulas and agent arguments
 - Move to 0.7.0 ostis-web-platform
+- Replaced ms_context with m_memoryCtx in DIA
```
Comment (MksmOrlov) [inline docs/changelog.md:32]: What is the difference? What bug is it entails

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — APPROVED** [review]

**Comment by MksmOrlov** [comment]
Also ask @NikitaZotov for review

File: problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp [hunk 3]
```diff
@@ -158,10 +167,17 @@ TEST_F(TemplateSearchManagerTest, SearchWithoutContent_MultipleResultTestCase)
       templateSearcher.searchTemplate(searchTemplateAddr, templateParams);
 
   EXPECT_TRUE(searchResults.size() == 2);
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp:169]: Use EXPECT_EQ instead

File: problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp [hunk 4]
```diff
@@ -11,6 +11,8 @@
 #include "searcher/TemplateSearcher.hpp"
 #include "keynodes/InferenceKeynodes.hpp"
 
+#include "algorithm"
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp:14]:
```suggestion
#include <algorithm>
```

File: problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp [hunk 5]
```diff
@@ -97,10 +99,17 @@ TEST_F(TemplateSearchManagerTest, SearchWithContent_MultipleResultTestCase)
       templateSearcher.searchTemplate(searchTemplateAddr, templateParams);
 
   EXPECT_TRUE(searchResults.size() == 2);
-  EXPECT_TRUE(
-      searchResults[1][searchLinkIdentifier] == context.HelperFindBySystemIdtf(firstCorrectResultLinkIdentifier));
-  EXPECT_TRUE(
-      searchResults[0][searchLinkIdentifier] == context.HelperFindBySystemIdtf(secondCorrectResultLinkIdentifier));
+  ScAddrVector expectedResults = {
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/test/units/TestTemplateSearchModule.cpp:102]: Use ScAddrSet instead

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**